Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/EndRideAfterThresholdTimePassed.hs

```diff
@@ -0,0 +1,31 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+module SharedLogic.Allocator.Jobs.EndRideAfterThresholdTimePassed where
+
+import qualified Domain.Action.UI.Ride.EndRide as RideEnd
+import Environment (AppEnv)
+import Kernel.Prelude hiding (handle)
+import Kernel.Types.Flow (FlowR)
+import Lib.Scheduler
+import SharedLogic.Allocator (AllocatorJobType (..))
+
+endRideAfterThresholdTimePassed ::
+  Job 'EndRideAfterThresholdTimePassed ->
+  FlowR AppEnv ExecutionResult
+endRideAfterThresholdTimePassed Job {jobInfo} = do
+  let jobData = jobInfo.jobData
+  let rideId = jobData.rideId
+  let merchantId = jobData.merchantId
+  _ <- RideEnd.buildEndRideHandle merchantId >>= (\hndl -> RideEnd.scheduleBasedEndRide hndl rideId (RideEnd.ScheduledBasedEndRideReq rideId))
```

Comment:
No need to call this if the ride has already ended. Can you add that check here?